packages/teams-js/test/internal/communication.spec.ts

```diff
@@ -1,67 +1,378 @@
 import * as communication from '../../src/internal/communication';
+import { GlobalVars } from '../../src/internal/globalVars';
 import { DOMMessageEvent } from '../../src/internal/interfaces';
 import * as utils from '../../src/internal/utils';
+import { FrameContexts } from '../../src/public';
+import { app } from '../../src/public/app';
+import { Utils } from '../utils';
 
-/* eslint-disable */
-/* As part of enabling eslint on test files, we need to disable eslint checking on the specific files with
-   large numbers of errors. Then, over time, we can fix the errors and reenable eslint on a per file basis. */
+describe('Testing communication', () => {
+  describe('initializeCommunication', () => {
+    describe('frameless', () => {
+      let utils: Utils = new Utils();
 
-describe('communication', () => {
-  it('initializeCommunication should reject if no parent window and no native interface found', async () => {
-    const initPromise = communication.initializeCommunication(undefined);
-    await expect(initPromise).rejects.toThrowError('Initialization Failed. No Parent window found.');
-  });
-  it('processMessage fail if message has a missing data property', () => {
-    const event = { badData: '' } as any as DOMMessageEvent;
-    const result = communication.processMessage(event);
-    expect(result).toBeUndefined();
-  });
-  it('processMessage fail if message is empty', () => {
-    const event = {} as any as DOMMessageEvent;
-    const result = communication.processMessage(event);
-    expect(result).toBeUndefined();
-  });
-  it('processMessage fail if data property is not an object', () => {
-    const event = { data: '' } as any as DOMMessageEvent;
-    const result = communication.processMessage(event);
-    expect(result).toBeUndefined();
-  });
-  it('processMessage fail if message has random data', () => {
-    const event = { badData: '', notAnOrigin: 'blah' } as any as DOMMessageEvent;
-    const result = communication.processMessage(event);
-    expect(result).toBeUndefined();
-  });
-  it('processMessage fail if data is undefined', () => {
-    const event = { data: undefined } as any as DOMMessageEvent;
-    const result = communication.processMessage(event);
-    expect(result).toBeUndefined();
-  });
-  it('processMessage fail if data is null', () => {
-    const event = { data: null } as any as DOMMessageEvent;
-    const result = communication.processMessage(event);
-    expect(result).toBeUndefined();
-  });
-  it('shouldProcessMessage fail if message source is same window ', () => {
-    communication.Communication.currentWindow = window;
-    // window object should now equal Communication.currentWindow
-    const result = communication.shouldProcessMessage(window, 'testOrigin.com');
-    expect(result).toBe(false);
+      beforeEach(() => {
+        // Set a mock window for testing
+        utils = new Utils();
```

Comment:
I might be wrong but why are we initializing `Utils` class again?

Reply:
You are correct that I am initializing Utils at the beginning of each unit test. Utils has internal state and the tests sometimes make changes to it. Creating a new instance of the Utils class just guarantees that each test isn't interfering with any of the other tests by accidentally leaving behind some change it made to the Utils object during test execution

Reply:
Shreyas might have been asking why we initialize it once on line 12 and then again before every unit test?

(to which I think the answer is: we have to declare the utils instance at the "describe" level in order for it to be accessible to every test. We have to initialize it when we declare it because if we don't initialize it there, that means it has to be typed as `Utils | null` and we don't want that because it means every test would have to be written to ensure that it's not `null`)

(he also might have not been asking that at all)

Reply:
Yes, if that is what Shreyas was asking then what Aaron said is why I did it. I turned on linting in this unit test file and fixed all the linting errors because why not? When I did that, I could no longer declare utils without initializing it without making its type Utils | undefined.